custom_components/wibeee/nest.py

```diff
@@ -38,11 +45,12 @@ def dispatch(self, mac_addr: str, pushed_data: Dict):
 @singleton.singleton("wibeee_nest_proxy")
 async def get_nest_proxy(
         hass: HomeAssistant,
-        upstream='http://nest-ingest.wibeee.com',
-        local_port=8600,
+        entry: ConfigEntry
```

Comment:
This is a `@singleton`, which means that there is only a single instance for all of Home Assistant. The decorator expects that the factory def must only take a single param of type `HomeAssistant` (you can [see how it's implemented here](https://github.com/luuuis/home-assistant-core/blob/dev/homeassistant/helpers/singleton.py#L38-L52)) and that is the reason why you're getting `TypeError: get_nest_proxy() takes 1 positional argument but 2 were given`..

Basically it will not be possible to have a singleton per `ConfigEntry` and probably you don't want an instance per `ConfigEntry` either (too many ports used and unnecessary).

Reply:
ok, I understand